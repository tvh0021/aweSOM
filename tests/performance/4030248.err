slurmstepd: error: *** JOB 4030248 ON worker5105 CANCELLED AT 2024-09-28T00:56:46 DUE TO TIME LIMIT ***
