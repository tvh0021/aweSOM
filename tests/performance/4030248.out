Benchmarking POPSOM with 1000 points
Training POPSOM
Training done
Time taken:  4.540255069732666
Computing U-matrix
U-matrix computed
Time taken:  0.001085519790649414
Computing SOM projection
SOM projection computed
Time taken:  0.0015881061553955078
Computing centroids
Number of clusters
6
Centroids computed
Time taken:  0.00136566162109375
Computing cluster labels
0
100
200
300
400
500
600
700
800
900
Cluster labels computed
Time taken:  0.003209352493286133
Total time taken:  4.547520637512207
Benchmarking aweSOM with 1000 points
Training aweSOM
starting epoch is:  0
stopping epoch is:  1000
Saving lattice every  5  epochs
Begin training
Evaluating epoch =  0
Evaluating epoch =  100
Evaluating epoch =  200
Evaluating epoch =  300
Evaluating epoch =  400
Evaluating epoch =  500
Evaluating epoch =  600
Evaluating epoch =  700
Shrinking neighborhood size to 8 at epoch 750
Evaluating epoch =  800
Evaluating epoch =  900
Evaluating epoch =  1000
Terminating from step limit reached at epoch  1000
Training complete
Training done
Time taken:  5.775596857070923
Computing U-matrix
U-matrix computed
Time taken:  0.0003161430358886719
Computing SOM projection and centroids
Begin matching points with nodes
i =  0
i =  200
i =  400
i =  800
i =  300
i =  500
i =  600
i =  700
i =  900
i =  100
Number of unique centroids:  4
Minimum cost between centroids:  0.2212867208720284
Number of clusters : 4
Centroids:  [(np.int64(5), np.int64(3)), (np.int64(0), np.int64(3)), (np.int64(7), np.int64(1)), (np.int64(0), np.int64(0))]
SOM projection and centroids computed
Time taken:  3.8834340572357178
Computing cluster labels
Cluster labels computed
Time taken:  0.6444363594055176
Total time taken:  10.30380392074585
Ratio of time taken by aweSOM to POPSOM for N = 1000, F = 2 : 2.2658069620949997
---------------------------------------------------
Benchmarking POPSOM with 10000 points
Training POPSOM
Training done
Time taken:  4.384631156921387
Computing U-matrix
U-matrix computed
Time taken:  0.001722097396850586
Computing SOM projection
SOM projection computed
Time taken:  0.01089620590209961
Computing centroids
Number of clusters
10
Centroids computed
Time taken:  0.007719278335571289
Computing cluster labels
0
1000
2000
3000
4000
5000
6000
7000
8000
9000
Cluster labels computed
Time taken:  0.03154444694519043
Total time taken:  4.436529874801636
Benchmarking aweSOM with 10000 points
Training aweSOM
starting epoch is:  0
stopping epoch is:  10000
Saving lattice every  50  epochs
Begin training
Evaluating epoch =  0
Evaluating epoch =  1000
Evaluating epoch =  2000
Evaluating epoch =  3000
Evaluating epoch =  4000
Evaluating epoch =  5000
Shrinking neighborhood size to 16 at epoch 5304
Shrinking neighborhood size to 15 at epoch 5616
Shrinking neighborhood size to 14 at epoch 5928
Evaluating epoch =  6000
Shrinking neighborhood size to 13 at epoch 6240
Shrinking neighborhood size to 12 at epoch 6552
Shrinking neighborhood size to 11 at epoch 6864
Evaluating epoch =  7000
Shrinking neighborhood size to 10 at epoch 7176
Shrinking neighborhood size to 9 at epoch 7488
Shrinking neighborhood size to 8 at epoch 7800
Evaluating epoch =  8000
Evaluating epoch =  9000
Evaluating epoch =  10000
Terminating from step limit reached at epoch  10000
Training complete
Training done
Time taken:  0.3651096820831299
Computing U-matrix
U-matrix computed
Time taken:  0.0005452632904052734
Computing SOM projection and centroids
Begin matching points with nodes
i =  0
i =  9000
i =  1000
i =  2000
i =  3000
i =  8000
i =  5000
i =  6000
i =  7000
i =  4000
Number of unique centroids:  8
Minimum cost between centroids:  0.0699680143401791
Number of clusters : 8
Centroids:  [(np.int64(9), np.int64(0)), (np.int64(0), np.int64(0)), (np.int64(12), np.int64(0)), (np.int64(0), np.int64(6)), (np.int64(14), np.int64(6)), (np.int64(14), np.int64(0)), (np.int64(6), np.int64(6)), (np.int64(14), np.int64(2))]
SOM projection and centroids computed
Time taken:  0.04646873474121094
Computing cluster labels
Cluster labels computed
Time taken:  0.008231401443481445
Total time taken:  0.4203810691833496
Ratio of time taken by aweSOM to POPSOM for N = 10000, F = 2 : 0.09475447727085248
---------------------------------------------------
Benchmarking POPSOM with 100000 points
Training POPSOM
Training done
Time taken:  53.82742619514465
Computing U-matrix
U-matrix computed
Time taken:  0.0062906742095947266
Computing SOM projection
SOM projection computed
Time taken:  0.10007810592651367
Computing centroids
Number of clusters
35
Centroids computed
Time taken:  0.10002923011779785
Computing cluster labels
0
10000
20000
30000
40000
50000
60000
70000
80000
90000
Cluster labels computed
Time taken:  0.314788818359375
Total time taken:  54.34863090515137
Benchmarking aweSOM with 100000 points
Training aweSOM
starting epoch is:  0
stopping epoch is:  100000
Saving lattice every  500  epochs
Begin training
Evaluating epoch =  0
Evaluating epoch =  10000
Evaluating epoch =  20000
Evaluating epoch =  30000
Evaluating epoch =  40000
Evaluating epoch =  50000
Shrinking neighborhood size to 27 at epoch 51250
Shrinking neighborhood size to 26 at epoch 52500
Shrinking neighborhood size to 25 at epoch 53750
Shrinking neighborhood size to 24 at epoch 55000
Shrinking neighborhood size to 23 at epoch 56250
Shrinking neighborhood size to 22 at epoch 57500
Shrinking neighborhood size to 21 at epoch 58750
Evaluating epoch =  60000
Shrinking neighborhood size to 20 at epoch 60000
Shrinking neighborhood size to 19 at epoch 61250
Shrinking neighborhood size to 18 at epoch 62500
Shrinking neighborhood size to 17 at epoch 63750
Shrinking neighborhood size to 16 at epoch 65000
Shrinking neighborhood size to 15 at epoch 66250
Shrinking neighborhood size to 14 at epoch 67500
Shrinking neighborhood size to 13 at epoch 68750
Evaluating epoch =  70000
Shrinking neighborhood size to 12 at epoch 70000
Shrinking neighborhood size to 11 at epoch 71250
Shrinking neighborhood size to 10 at epoch 72500
Shrinking neighborhood size to 9 at epoch 73750
Shrinking neighborhood size to 8 at epoch 75000
Evaluating epoch =  80000
Evaluating epoch =  90000
Evaluating epoch =  100000
Terminating from step limit reached at epoch  100000
Training complete
Training done
Time taken:  4.4452784061431885
Computing U-matrix
U-matrix computed
Time taken:  0.001718759536743164
Computing SOM projection and centroids
Begin matching points with nodes
i =  0
i =  60000
i =  10000
i =  50000
i =  90000
i =  80000
i =  30000
i =  20000
i =  70000
i =  40000
Number of unique centroids:  9
Minimum cost between centroids:  0.042535157075515556
Number of clusters : 9
Centroids:  [(np.int64(1), np.int64(12)), (np.int64(21), np.int64(0)), (np.int64(0), np.int64(0)), (np.int64(26), np.int64(12)), (np.int64(8), np.int64(0)), (np.int64(14), np.int64(0)), (np.int64(26), np.int64(0)), (np.int64(13), np.int64(12)), (np.int64(25), np.int64(7))]
SOM projection and centroids computed
Time taken:  0.060860395431518555
Computing cluster labels
Cluster labels computed
Time taken:  0.006094694137573242
Total time taken:  4.513967275619507
Ratio of time taken by aweSOM to POPSOM for N = 100000, F = 2 : 0.08305576792720377
---------------------------------------------------
Benchmarking POPSOM with 1000000 points
Training POPSOM
Training done
Time taken:  1464.6081900596619
Computing U-matrix
U-matrix computed
Time taken:  0.0425107479095459
Computing SOM projection
SOM projection computed
Time taken:  1.0349836349487305
Computing centroids
Number of clusters
97
Centroids computed
Time taken:  1.3143179416656494
Computing cluster labels
0
100000
200000
300000
400000
500000
600000
700000
800000
900000
Cluster labels computed
Time taken:  3.162984848022461
Total time taken:  1470.1630160808563
Benchmarking aweSOM with 1000000 points
Training aweSOM
starting epoch is:  0
stopping epoch is:  1000000
Saving lattice every  5000  epochs
Begin training
Evaluating epoch =  0
Evaluating epoch =  100000
Evaluating epoch =  200000
Evaluating epoch =  300000
Evaluating epoch =  400000
Evaluating epoch =  500000
Shrinking neighborhood size to 49 at epoch 506051
Shrinking neighborhood size to 48 at epoch 512148
Shrinking neighborhood size to 47 at epoch 518245
Shrinking neighborhood size to 46 at epoch 524342
Shrinking neighborhood size to 45 at epoch 530439
Shrinking neighborhood size to 44 at epoch 536536
Shrinking neighborhood size to 43 at epoch 542633
Shrinking neighborhood size to 42 at epoch 548730
Shrinking neighborhood size to 41 at epoch 554827
Shrinking neighborhood size to 40 at epoch 560924
Shrinking neighborhood size to 39 at epoch 567021
Shrinking neighborhood size to 38 at epoch 573118
Shrinking neighborhood size to 37 at epoch 579215
Shrinking neighborhood size to 36 at epoch 585312
Shrinking neighborhood size to 35 at epoch 591409
Shrinking neighborhood size to 34 at epoch 597506
Evaluating epoch =  600000
Shrinking neighborhood size to 33 at epoch 603603
Shrinking neighborhood size to 32 at epoch 609700
Shrinking neighborhood size to 31 at epoch 615797
Shrinking neighborhood size to 30 at epoch 621894
Shrinking neighborhood size to 29 at epoch 627991
Shrinking neighborhood size to 28 at epoch 634088
Shrinking neighborhood size to 27 at epoch 640185
Shrinking neighborhood size to 26 at epoch 646282
Shrinking neighborhood size to 25 at epoch 652379
Shrinking neighborhood size to 24 at epoch 658476
Shrinking neighborhood size to 23 at epoch 664573
Shrinking neighborhood size to 22 at epoch 670670
Shrinking neighborhood size to 21 at epoch 676767
Shrinking neighborhood size to 20 at epoch 682864
Shrinking neighborhood size to 19 at epoch 688961
Shrinking neighborhood size to 18 at epoch 695058
Evaluating epoch =  700000
Shrinking neighborhood size to 17 at epoch 701155
Shrinking neighborhood size to 16 at epoch 707252
Shrinking neighborhood size to 15 at epoch 713349
Shrinking neighborhood size to 14 at epoch 719446
Shrinking neighborhood size to 13 at epoch 725543
Shrinking neighborhood size to 12 at epoch 731640
Shrinking neighborhood size to 11 at epoch 737737
Shrinking neighborhood size to 10 at epoch 743834
Shrinking neighborhood size to 9 at epoch 749931
Shrinking neighborhood size to 8 at epoch 756028
Evaluating epoch =  800000
Evaluating epoch =  900000
Evaluating epoch =  1000000
Terminating from step limit reached at epoch  1000000
Training complete
Training done
Time taken:  96.23142385482788
Computing U-matrix
U-matrix computed
Time taken:  0.03483176231384277
Computing SOM projection and centroids
Begin matching points with nodes
i =  900000
i =  400000
i =  800000
i =  200000
i =  700000
i =  300000
i =  500000
i =  0
i =  600000
i =  100000
Number of unique centroids:  34
Minimum cost between centroids:  0.009909775750229782
Number of clusters : 34
Centroids:  [(np.int64(23), np.int64(4)), (np.int64(6), np.int64(15)), (np.int64(40), np.int64(23)), (np.int64(37), np.int64(9)), (np.int64(21), np.int64(22)), (np.int64(47), np.int64(10)), (np.int64(17), np.int64(12)), (np.int64(10), np.int64(0)), (np.int64(43), np.int64(0)), (np.int64(0), np.int64(8)), (np.int64(25), np.int64(7)), (np.int64(0), np.int64(23)), (np.int64(18), np.int64(7)), (np.int64(31), np.int64(16)), (np.int64(12), np.int64(21)), (np.int64(28), np.int64(14)), (np.int64(25), np.int64(0)), (np.int64(43), np.int64(5)), (np.int64(16), np.int64(0)), (np.int64(7), np.int64(0)), (np.int64(20), np.int64(0)), (np.int64(26), np.int64(16)), (np.int64(34), np.int64(2)), (np.int64(45), np.int64(17)), (np.int64(0), np.int64(0)), (np.int64(3), np.int64(23)), (np.int64(14), np.int64(23)), (np.int64(36), np.int64(23)), (np.int64(47), np.int64(23)), (np.int64(46), np.int64(0)), (np.int64(9), np.int64(21)), (np.int64(0), np.int64(15)), (np.int64(7), np.int64(5)), (np.int64(38), np.int64(17))]
SOM projection and centroids computed
Time taken:  0.32395267486572266
Computing cluster labels
Cluster labels computed
Time taken:  0.006501674652099609
Total time taken:  96.59672594070435
Ratio of time taken by aweSOM to POPSOM for N = 1000000, F = 2 : 0.06570477211310266
---------------------------------------------------
Benchmarking POPSOM with 10000000 points
Training POPSOM
